lib/liquid.lua

```diff
@@ -2547,28 +2549,39 @@ end
 -------------------------------------------------------------ParserContext end ---------------------------------------------------------
 -------------------------------------------------------------FileSystem begin ---------------------------------------------------------
 -- local FileSystem = {}
-function FileSystem:new( location )
+function FileSystem:new( get, error_handler )
     -- body
     local instance = {}
     setmetatable(instance, {__index = FileSystem})
-    instance.location = location
+    instance.get = get
     instance.text = nil
+    instance.error_handler = error_handler
     return instance
 end
 --
-function FileSystem:genertic_get( ... )
+function FileSystem:genertic_get( location )
+    local error_handler = self.error_handler
     -- body
     if self.get and type(self.get) == "function" then
-        local status, err = pcall(self.get, self.location)
-        if status then
-            return err
+        local ok, ret = pcall(self.get, location)
+
+        if not ret then
```

Comment:
1. how about using the condition `if type(ret) ~= "sting" then `?

2. the `status` should be checked before the the `ret` . 
i.e.
```
if ok then 
    return ret
else
   -- `ret` checking should be here
 end
```
you can simplely move these lines(2568-2570) to line 2575.